File: ui/src/app/model/loja.ts
import {FormControl, FormGroup, Validators} from '@angular/forms';
import {TipoPessoa} from './tipo-pessoa';
import {Validador} from '../core/validador';
import * as moment from 'moment';

export class Loja {

  id: number | undefined;
  nome: string | undefined;
  tipoPessoa: TipoPessoa | undefined;

  cpf: string | undefined;

  cnpj: string | undefined;
  razaoSocial: string | undefined;
  dataFundacao: string | undefined;
  optanteSimplesNacional: boolean | undefined;
  vendeProduto: boolean | undefined;
  vendeServico: boolean | undefined;
  inscricaoMunicipal: string | undefined;
  inscricaoEstadual: string | undefined;
  isentoInscricaoEstadual: boolean | undefined;

  cep: string | undefined;
  endereco: string | undefined;
  numero: string | undefined;
  complemento: string | undefined;
  bairro: string | undefined;
  cidade: string | undefined;
  uf: string | undefined;
  ibge: string | undefined;
  gia: string | undefined;
  ddd: string | undefined;
  siafi: string | undefined;
  telefoneComercial: string | undefined;
  email: string | undefined;

  constructor() {
    this.tipoPessoa = TipoPessoa.JURIDICA;
    this.vendeProduto = true;
    this.vendeServico = true;
    this.optanteSimplesNacional = true;
    this.isentoInscricaoEstadual = true;
    this.dataFundacao = moment().format('YYYY-MM-DD');
  }

  static criarFormulario(loja: Loja) {

    return new FormGroup({
      id: new FormControl(loja.id),
      tipoPessoa: new FormControl(loja.tipoPessoa),
      nome: new FormControl(loja.nome, Validators.required),
      cpf: new FormControl(loja.cpf, Validador.cpfObrigatorio()),
      cnpj: new FormControl(loja.cnpj, Validador.cnpjObrigatorio()),
      razaoSocial: new FormControl(loja.razaoSocial, [Validators.required]),
      dataFundacao: new FormControl(loja.dataFundacao),
      optanteSimplesNacional: new FormControl(loja.optanteSimplesNacional),
      vendeProduto: new FormControl(loja.vendeProduto),
      vendeServico: new FormControl(loja.vendeServico),
      inscricaoMunicipal: new FormControl(loja.inscricaoMunicipal),
      inscricaoEstadual: new FormControl(loja.inscricaoEstadual),
      isentoInscricaoEstadual: new FormControl(loja.isentoInscricaoEstadual),
      cep: new FormControl(loja.cep),
      endereco: new FormControl(loja.endereco),
      numero: new FormControl(loja.numero),
      complemento: new FormControl(loja.complemento),
      bairro: new FormControl(loja.bairro),
      cidade: new FormControl(loja.cidade),
      uf: new FormControl(loja.uf),
      ibge: new FormControl(loja.ibge),
      gia: new FormControl(loja.gia),
      ddd: new FormControl(loja.ddd),
      siafi: new FormControl(loja.siafi),
      telefoneComercial: new FormControl(loja.telefoneComercial),
      email: new FormControl(loja.email, Validators.email)
    });
  }
}

File: api/src/main/java/com/example/demo/model/Loja.java
package com.example.demo.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDate;

@Data
@Entity
public class Loja {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(name = "tipo_pessoa")
    private TipoPessoa tipoPessoa;

    @Column(name = "nome")
    private String nome;

    // FISICA
    @Column(name = "cpf")
    private String cpf;

    // JURIDICA
    @Column(name = "cnpj")
    private String cnpj;

    @Column(name = "razao_social")
    private String razaoSocial;

    @Column(name = "data_fundacao")
    private LocalDate dataFundacao;

    @Column(name = "optante_simples_nacional")
    private boolean optanteSimplesNacional;

    @Column(name = "vende_produto")
    private boolean vendeProduto;

    @Column(name = "vende_servico")
    private boolean vendeServico;

    @Column(name = "inscricao_municipal")
    private String inscricaoMunicipal;

    @Column(name = "inscricao_estadual")
    private String inscricaoEstadual;

    @Column(name = "isento_inscricao_estadual")
    private boolean isentoInscricaoEstadual;

    // ENDERECO E CONTATOS
    @Column(name = "cep")
    private String cep;

    @Column(name = "endereco")
    private String endereco;

    @Column(name = "numero")
    private String numero;

    @Column(name = "complemento")
    private String complemento;

    @Column(name = "bairro")
    private String bairro;

    @Column(name = "cidade")
    private String cidade;

    @Column(name = "uf")
    private String uf;

    @Column(name = "ibge")
    private String ibge;

    @Column(name = "gia")
    private String gia;

    @Column(name = "ddd")
    private String ddd;

    @Column(name = "siafi")
    private String siafi;

    @Column(name = "telefone_comercial")
    private String telefoneComercial;

    @Column(name = "email")
    private String email;
}
